### PR TITLE
Move files from `protocol` to appropriate module

### PR DESCRIPTION
**Pull request by da-kami**
Some network and application specific code does not belong in the protocol module and was moved.
Eventloop, recovery and the outside behaviour were moved to the respective application module because they are application specific.

The `swap_setup` was moved into the network module because upon change both sides will have to be changed and should thus stay close together.

### CODE REVIEW

File: swap/src/asb.rs [hunk 1]
```diff
@@ -1,7 +1,19 @@
+pub use rate::Rate;
+
+pub mod behaviour;
 pub mod command;
 pub mod config;
+pub mod event_loop;
 mod rate;
+pub mod recovery;
```
Comment (thomaseizinger) [inline swap/src/asb.rs:8]: If we re-export, can these be private now?

File: swap/src/asb.rs [hunk 2]
```diff
@@ -1,7 +1,19 @@
+pub use rate::Rate;
```
Comment (thomaseizinger) [inline swap/src/asb.rs:1]: Should be part of the other `pub use` statements.

File: swap/src/bin/asb.rs [hunk 3]
```diff
@@ -26,12 +26,13 @@ use swap::asb::command::{parse_args, Arguments, Command};
 use swap::asb::config::{
     initial_setup, query_user_for_initial_config, read_config, Config, ConfigNotInitialized,
 };
+use swap::asb::event_loop::{EventLoop, KrakenRate};
+use swap::asb::recovery::redeem;
```
Comment (thomaseizinger) [inline swap/src/bin/asb.rs:30]: You are importing from the submodule even though we re-export. Making these modules private should allow us to be more consistent here.

File: swap/src/bin/swap.rs [hunk 4]
```diff
@@ -12,29 +12,32 @@
...
-use prettytable::{row, Table};
-use qrcode::render::unicode;
-use qrcode::QrCode;
 use std::cmp::min;
 use std::env;
 use std::future::Future;
 use std::path::PathBuf;
 use std::sync::Arc;
 use std::time::Duration;
+
```
Comment (thomaseizinger) [inline swap/src/bin/swap.rs:21]: Can we not have these newlines inbetween the use statements?

File: swap/src/bin/swap.rs [hunk 5]
```diff
@@ -428,12 +431,15 @@ where
...
-    use crate::determine_btc_to_swap;
-    use ::bitcoin::Amount;
     use std::sync::Mutex;
+
+    use ::bitcoin::Amount;
     use tracing::subscriber;
 
+    use crate::determine_btc_to_swap;
+
+    use super::*;
```
Comment (thomaseizinger) [inline swap/src/bin/swap.rs:441]: Same here. Are you using "organize imports" from CLion? I saw that causing this at some point.

File: swap/src/cli/behaviour.rs [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
+use std::time::Duration;
+
 use anyhow::{anyhow, Error, Result};
 use libp2p::core::Multiaddr;
 use libp2p::ping::{Ping, PingEvent};
 use libp2p::request_response::{RequestId, ResponseChannel};
 use libp2p::{NetworkBehaviour, PeerId};
-use std::sync::Arc;
-use std::time::Duration;
+
```
Comment (thomaseizinger) [inline swap/src/cli/behaviour.rs:9]: Again here.

File: swap/src/network/redial.rs [hunk 7]
```diff
@@ -7,12 +10,11 @@ use libp2p::core::Multiaddr;
 use libp2p::swarm::protocols_handler::DummyProtocolsHandler;
 use libp2p::swarm::{DialPeerCondition, NetworkBehaviour, NetworkBehaviourAction, PollParameters};
 use libp2p::PeerId;
-use std::pin::Pin;
-use std::task::{Context, Poll};
-use std::time::Duration;
 use tokio::time::{Instant, Sleep};
 use void::Void;
```
Comment (thomaseizinger) [inline swap/src/network/redial.rs:15]: Also here.

File: swap/src/network/swap_setup.rs [hunk 8]
```diff
@@ -1,10 +1,14 @@
-use crate::monero;
 use anyhow::{Context, Result};
 use libp2p::core::upgrade;
 use libp2p::swarm::NegotiatedSubstream;
 use serde::de::DeserializeOwned;
 use serde::{Deserialize, Serialize};
```
Comment (thomaseizinger) [inline swap/src/network/swap_setup.rs:6]: And here.

**Review by thomaseizinger — APPROVED** [review]
Looks good overall, I think making these modules private would help with consistency.

**Comment by da-kami** [comment]
I managed to make some of the modules private. For the `cli`'s `cancel` and `refund` it did not make much sense, because I would have to re-export the complete module again (because we have `Error` in both that are used within the same file - could work around that, but not sure it's worth it...)

**Comment by da-kami** [comment]
I'd merge this into the underlying branch first, then squash the fixups and then merge the underlying branch. Will do that Monday morning :)

**Review by thomaseizinger — APPROVED** [review]

**Comment by da-kami** [comment]
changes merged into #586

**Pull request closed without merging**